code/modules/grab/_grab.dm

```diff
@@ -1,430 +1,430 @@
+					else
+						dislocate(affecting, assailant, hit_zone)
+
+				if(I_DISARM)
+					pin_down(affecting, assailant)
+
+	//clicking on yourself while grabbing them
+	if(M == assailant && state >= GRAB_AGGRESSIVE)
+		if(assailant.devour(affecting))
+			qdel(src)
+
+/obj/item/grab/dropped()
+	..()
+	loc = null
+	if(!destroying)
+		qdel(src)
+
+/obj/item/grab/proc/reset_kill_state()
+	if(state == GRAB_KILL)
+		assailant.visible_message("<span class='warning'>[assailant] lost \his tight grip on [affecting]'s neck!</span>")
+		state = GRAB_NECK
+
+/obj/item/grab/proc/handle_resist()
+
+	if(!affecting)
+		var/mob/M = loc
+		if(istype(M))
+			M.drop_from_inventory(src)
+			return
+
+	if(!affecting_mob)
+		return // A crate is not going to resist.
+
+	var/grab_name = "grip"
+	var/break_strength = 1
+	var/list/break_chance_table = list(100)
+
+	switch(state)
+		if(GRAB_PASSIVE)
+			//Being knocked down makes it harder to break a grab, so it is easier to cuff someone who is down without forcing them into unconsciousness.
+			//use same chance_table as aggressive but give +2 for not-weakened so that corvid grabs don't become auto-success for weakened either, that's lame
+			if(!affecting_mob.incapacitated(INCAPACITATION_KNOCKDOWN))
+				break_strength += 2
+			break_chance_table = list(15, 60, 100)
+
+		if(GRAB_AGGRESSIVE)
+			//Being knocked down makes it harder to break a grab, so it is easier to cuff someone who is down without forcing them into unconsciousness.
+			if(!affecting_mob.incapacitated(INCAPACITATION_KNOCKDOWN))
+				break_strength++
+			break_chance_table = list(15, 60, 100)
+
+		if(GRAB_NECK)
+			grab_name = "headlock"
+			//If the you move when grabbing someone then it's easier for them to break free. Same if the affected mob is immune to stun.
+			if(world.time - assailant.l_move_time < 30 || !affecting_mob.stunned)
+				break_strength++
+			break_chance_table = list(3, 18, 45, 100)
+
+	//It's easier to break out of a grab by a smaller mob
+	break_strength += max(size_difference(affecting_mob, assailant), 0)
+
+	var/break_chance = break_chance_table[Clamp(break_strength, 1, break_chance_table.len)]
+	if(prob(break_chance))
+		if(grab_name)
+			affecting_mob.visible_message("<span class='warning'>\The [affecting_mob] has broken free of [assailant]'s [grab_name]!</span>")
+		qdel(src)
+
+//returns the number of size categories between affecting_mob and assailant, rounded. Positive means A is larger than B
+/obj/item/grab/proc/size_difference(mob/A, mob/B)
+	return mob_size_difference(A.mob_size, B.mob_size)
+
+/obj/item/grab
+	var/destroying = 0
+
+/obj/item/grab/Destroy()
+	affecting_mob = null
+	if(assailant_robot)
+		if(assailant_robot.current_grab == src)
+			assailant_robot.current_grab = null
+		assailant_robot = null
+	if(affecting)
+		reset_position()
+		LAZYREMOVE(affecting.grabbed_by, src)
+		affecting = null
+	if(assailant)
+		assailant = null
+	destroying = 1 // stops us calling qdel(src) on dropped()
```

Comment:
might as well just replace references to this with `if (QDELING(src))`, since the GC tracks this anyways.